src/Codeception/Module/Redis.php

```diff
@@ -3,87 +3,626 @@
+                $this->driver->set($key, $value);
+                break;
+
+            case 'list':
+                $this->driver->rpush($key, $value);
+                break;
+
+            case 'set':
+                $this->driver->sadd($key, $value);
+                break;
+
+            case 'zset':
+                if (!is_array($value)) {
+                    throw new ModuleException(
+                        $this,
+                        'If second argument of haveInRedis() method is "zset", '
+                        . 'third argument must be an (associative) array'
+                    );
+                }
+                $this->driver->zadd($key, $value);
+                break;
+
+            case 'hash':
+                if (!is_array($value)) {
+                    throw new ModuleException(
+                        $this,
+                        'If second argument of haveInRedis() method is "hash", '
+                        . 'third argument must be an array'
+                    );
+                }
+                $this->driver->hmset($key, $value);
+                break;
+
+            default:
+                throw new ModuleException(
+                    $this,
+                    "Unknown type \"$type\" for key \"$key\". Allowed types are "
+                    . '"string", "list", "set", "zset", "hash"'
+                );
         }
     }
+
+    /**
+     * Asserts that a key does not exist or, optionaly, that it doesn't have the
+     * provided $value
+     *
+     * Examples:
+     *
+     * ```php?start_inline=1
+     * // With only one argument, only checks the key does not exist
+     * $I->dontSeeInRedis('example:string');
+     * // Checks a String does not exist or its value is not the one provided
+     * $I->dontSeeInRedis('example:string', 'life');
+     * // Checks a List does not exist or its value is not the one provided (order of elements is compared).
+     * $I->dontSeeInRedis('example:list', ['riri', 'fifi', 'loulou'])
+     * // Checks a Set does not exist or its value is not the one provided (order of members is ignored).
+     * $I->dontSeeInRedis('example:set', ['riri', 'fifi', 'loulou'])
+     * // Checks a ZSet does not exist or its value is not the one provided (scores are required, order of members is compared)
+     * $I->dontSeeInRedis('example:zset', ['riri' => 1, 'fifi' => 2, 'loulou' => 3])
+     * // Checks a Hash does not exist or its value is not the one provided (order of members is ignored).
+     * $I->dontSeeInRedis('example:hash', ['riri' => true, 'fifi' => 'Dewey', 'loulou' => 2])
+     * ```
+     *
+     * @param string $key   The key name
+     * @param mixed  $value Optional. If specified, also checks the key has this
+     * value. Booleans will be converted to 1 and 0 (even inside arrays)
+     */
+    public function dontSeeInRedis($key, $value = null)
+    {
+        $this->assertFalse(
+            $this->checkKeyExists($key, $value)
+        );
```

Comment:
please add better failure message to this exception.
in current version user will see "failed asserting that true is false" which won't be really helpful